src/saving/ISaveUpgradeStep.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace Saving
+{
+    using System;
+    using System.Collections.Generic;
+    using Godot;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    public interface ISaveUpgradeStep
+    {
+        /// <summary>
+        ///   Performs a save upgrade. Throws exceptions on failure.
+        /// </summary>
+        /// <param name="saveInfo">Loaded save info from inputSave</param>
+        /// <param name="inputSave">Save name that can be used to read the save data</param>
+        /// <param name="outputSave">Target name of the upgraded save</param>
+        /// <returns>The new version of the upgraded save</returns>
+        string PerformUpgrade(SaveInformation saveInfo, string inputSave, string outputSave);
+
+        /// <summary>
+        ///   Calculates what the version of save would be if PerformUpgrade is ran on it
+        /// </summary>
+        /// <param name="saveInfo">The save info to inspect</param>
+        /// <returns>The version after upgrade, null if can't upgrade</returns>
+        string VersionAfterUpgrade(SaveInformation saveInfo);
+    }
+
+    public static class SaveUpgradeSteps
+    {
+        private static readonly Dictionary<string, ISaveUpgradeStep> StoredSaveUpgradeSteps =
+            InitializeSaveUpgradeSteps();
+
+        public static IReadOnlyDictionary<string, ISaveUpgradeStep> SupportedUpgrades => StoredSaveUpgradeSteps;
+
+        public static ISaveUpgradeStep GetUpgradeStepForVersion(string version)
+        {
+            if (!SupportedUpgrades.TryGetValue(version, out ISaveUpgradeStep step))
+                return null;
+
+            return step;
+        }
+
+        private static Dictionary<string, ISaveUpgradeStep> InitializeSaveUpgradeSteps()
+        {
+            // TODO: would it be useful to specify a range of versions an upgrader can upgrade to make it less error
+            // prone to accidentally miss a version
+            return new Dictionary<string, ISaveUpgradeStep>
+            {
+                { "0.5.4.0", new UpgradeStep054To055() },
+            };
+        }
+    }
+
+    internal class UpgradeStep054To055 : BaseJSONUpgradeStep
+    {
+        protected override string VersionAfter => "0.5.5.0-alpha";
+
+        protected override void PerformUpgradeOnJSON(JObject saveData)
+        {
+            foreach (var entry in saveData.Properties())
+            {
+                RecursivelyUpdateMembraneValues(entry);
+            }
+        }
+
+        private void RecursivelyUpdateMembraneValues(JProperty property)
+        {
+            if (property.Name.Contains("Membrane") || property.Name.Contains("membrane"))
```

Comment:
```suggestion
            if (property.Name.ToLower().Contains("membrane"))
```

Reply:
I'm quite torn on this one...
This will need to allocate a new converted string for *each* key in the save json (which can be multiple megabytes large), while also only marginally shortening this. I went with this approach as I only really considered there possibly being these two values possible.